string collection_name
time stamp
string record